test/realm_test.dart

```diff
@@ -1551,6 +1550,210 @@ Future<void> main([List<String>? args]) async {
+  });
+
+  baasTest('Realm writeCopy Local->Sync is not supported', (appConfiguration) async {
+    final originalConfig = Configuration.local([Product.schema]);
+    final originalRealm = getRealm(originalConfig);
+
+    final app = App(appConfiguration);
+    final credentials = Credentials.anonymous(reuseCredentials: false);
+    var user = await app.logIn(credentials);
+    final configCopy = Configuration.flexibleSync(user, [Product.schema]);
+    expect(() => originalRealm.writeCopy(configCopy),
+        throws<RealmException>("Realm cannot be converted to a flexible sync realm unless flexible sync is already enabled"));
+  });
+
+  test('Realm writeCopy Local->Local inside a write block', () {
+    final originalConfig = Configuration.local([Car.schema]);
+    final originalRealm = getRealm(originalConfig);
+    final pathCopy = originalConfig.path.replaceFirst(p.basenameWithoutExtension(originalConfig.path), generateRandomString(10));
+    final configCopy = Configuration.local([Car.schema], path: pathCopy);
+    var itemsCount = 10;
+    originalRealm.write(() {
+      for (var i = 0; i < itemsCount; i++) {
+        originalRealm.add(Car("make_${i + 1}"));
+      }
+      originalRealm.writeCopy(configCopy);
+    });
+    originalRealm.close();
+
+    expect(File(pathCopy).existsSync(), isTrue);
+    final copiedRealm = getRealm(configCopy);
+    expect(copiedRealm.all<Car>().length, itemsCount);
+    expect(copiedRealm.isClosed, isFalse);
+    copiedRealm.close();
+  });
+
+  test('Realm writeCopy Local->Local read-only', () {
+    final originalConfig = Configuration.local([Car.schema]);
+    final originalRealm = getRealm(originalConfig);
+    final pathCopy = originalConfig.path.replaceFirst(p.basenameWithoutExtension(originalConfig.path), generateRandomString(10));
+    final configCopy = Configuration.local([Car.schema], path: pathCopy, isReadOnly: true);
+
+    originalRealm.writeCopy(configCopy);
+    originalRealm.close();
+
+    expect(File(pathCopy).existsSync(), isTrue);
+    final copiedRealm = getRealm(configCopy);
+    expect(copiedRealm.isClosed, isFalse);
+    _expectAllWritesToThrow<RealmException>(copiedRealm, "Can't perform transactions on read-only Realms.");
+    _expectAllAsyncWritesToThrow<RealmException>(copiedRealm, "Can't perform transactions on read-only Realms.");
+
+    copiedRealm.close();
+  });
+
+  test('Realm writeCopy in-memory Local->Local', () {
+    final originalConfig = Configuration.inMemory([Car.schema]);
+    final originalRealm = getRealm(originalConfig);
+    final pathCopy = originalConfig.path.replaceFirst(p.basenameWithoutExtension(originalConfig.path), generateRandomString(10));
+    final configCopy = Configuration.local([Car.schema], path: pathCopy);
+
+    originalRealm.writeCopy(configCopy);
+    originalRealm.close();
+
+    expect(File(pathCopy).existsSync(), isTrue);
+    final copiedRealm = getRealm(configCopy);
+    expect(copiedRealm.isClosed, isFalse);
```

Comment:
why do we need this check exactly after we have opend a realm. That really not needed imo

Reply:
I usually add this to all the tests, which have to prove that the realm can be opened.

Reply:
Will remove that.

Reply:
> I usually add this to all the tests, which have to prove that the realm can be opened.
This should not be needed in any test other than the `isClosed` test that we have already